src/Microsoft.ML.Transforms/PermutationFeatureImportance.cs

```diff
@@ -0,0 +1,224 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.ML.Core.Data;
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.Api;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Runtime.Internal.Utilities;
+using Float = System.Single;
+
+namespace Microsoft.ML.Transforms
+{
+    using Stopwatch = System.Diagnostics.Stopwatch;
+
+    internal sealed class PermutationFeatureImportance<TResult>
+    {
+        private readonly IHostEnvironment _host;
+        private VBuffer<ReadOnlyMemory<char>> _slotNames;
+        private readonly List<TResult> _metricsDelta;
+
+        public PermutationFeatureImportance(IHostEnvironment host)
+        {
+            Contracts.CheckValue(host, nameof(host));
+            _host = host;
+            _metricsDelta = new List<TResult>();
+        }
+
+        public ImmutableArray<TResult>
+            GetImportanceMetricsMatrix(
+                ITransformer model,
+                IDataView data,
+                Func<IDataView, TResult> evaluationFunc,
+                Func<TResult, TResult, TResult> deltaFunc,
+                string features,
+                int topExamples,
+                int progressIterations)
```

Comment:
>progressIterations [](start = 20, length = 18)

There is no point in having this as a top-level parameter. Just print progress every 10th feature, that's all. Or every feature.